src/Uno.Extensions.Navigation.UI/RouteResolverDefault.cs

```diff
@@ -96,32 +94,43 @@ view is null &&
 		if (viewMap is not null)
 		{
 			var viewFunc = (viewMap.View is not null) ?
-										() => viewMap.View :
-										viewMap.ViewSelector;
+										   () => viewMap.View :
+										   viewMap.ViewSelector;
 			var defaultMapFromViewMap = new RouteInfo(
-												Path: path,
-												View: viewFunc,
-												ViewAttributes: viewMap.ViewAttributes,
-												ViewModel: viewMap.ViewModel,
-												Data: viewMap.Data?.Data,
-												ToQuery: viewMap?.Data?.UntypedToQuery,
-												FromQuery: viewMap?.Data?.UntypedFromQuery,
-												ResultData: viewMap?.ResultData,
-												IsDialogViewType: () =>
-												{
-													return IsDialogViewType(viewFunc?.Invoke());
-												});
+										   Path: path,
+										   View: viewFunc,
+										   ViewAttributes: viewMap.ViewAttributes,
+										   ViewModel: viewMap.ViewModel,
+										   Data: viewMap.Data?.Data,
+										   ToQuery: viewMap?.Data?.UntypedToQuery,
+										   FromQuery: viewMap?.Data?.UntypedFromQuery,
+										   ResultData: viewMap?.ResultData,
+										   IsDialogViewType: () =>
+										   {
+											   return IsDialogViewType(viewFunc?.Invoke());
+										   });
 			Mappings.Add(defaultMapFromViewMap);
 			if (Logger.IsEnabled(LogLevel.Debug)) Logger.LogDebugMessage($"Created default mapping from viewmap - Path '{defaultMapFromViewMap.Path}'");
 			return defaultMapFromViewMap;
 		}
 
-		if (Logger.IsEnabled(LogLevel.Information)) Logger.LogInformationMessage($"For better performance (avoid reflection), create mapping for for path '{path}', view '{view?.Name}', view model '{viewModel?.Name}'");
+		if (Logger.IsEnabled(LogLevel.Information))
+			Logger.LogInformationMessage($"For better performance (avoid reflection), create mapping for path '{path}', view '{view?.Name}', view model '{viewModel?.Name}'");
 
 		if (view is null)
 		{
 			var trimmedPath = TrimSuffices(path, ViewModelSuffixes);
 			view = TypeFromPath(trimmedPath, true, ViewSuffixes, type => type.IsSubclassOf(typeof(FrameworkElement)));
+
+			if (view is not null)
+			{
+				if (view != null)
+				{
```

Comment:
You don't need this check, you are already inside of the `view is not null` if block